common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilter.java

```diff
@@ -148,6 +153,24 @@ int getVersionNumber() {
   public abstract boolean mightContainBinary(byte[] item);
 
   /**
+   * Returns a new Bloom filter of the union of two Bloom filters.
+   * Unlike mergeInplace, this will not cause a mutation.
+   * Callers must ensure the bloom filters are appropriately sized to avoid saturating them.
+   *
+   * @param other The bloom filter to union this bloom filter with.
+   * @throws IncompatibleUnionException if {@code isCompatible(other) == false}
+   */
+  public abstract BloomFilterImpl union(BloomFilter other) throws IncompatibleUnionException;
+
+  /**
+   * Swamidass & Baldi (2007) approximation for number of items in the intersection of two Bloom filters
```

Comment:
* Same here. Document the method first and then mention the reference.
* How is it different from intersecting two bloom filters and then estimate the number of items? Union might lead to larger approximation error.